src/tests/uat/enriched_events_uat.py

```diff
@@ -0,0 +1,275 @@
+    while not future.done():
+        if print_detail:
+            pass
+
+    results = future.result()
+    pprint(results._info, sort_dicts=False)
+
+    print("\nCompare results manually with Postman")
+    print("----------------------------------------------------------")
+
+
+def enriched_events_details(cb, event_id):
+    """Start SDK call and print out response"""
+    print("API Calls:")
+    print("Request Details for Enriched Events (v2)")
+    print("Retrieve Results for an Enriched Event Detail Search (v2)")
+    print(f"event_id: {event_id}\n")
+
+    events = cb.select(EnrichedEvent).where(event_id=event_id)
+    pprint(events[0].get_details())
+
+    print("\nCompare results manually with Postman")
+    print("----------------------------------------------------------")
+
+
+def enriched_events_aggregation(cb):
+    """Start SDK call and print out response"""
+    print("API Calls:")
+    print("Start Aggregation Search on Enriched Events (v1)")
+    print("Retrieve Results for an Enriched Event Aggregation Search (v1)\n")
+
+    query = cb.select(EnrichedEvent).where("process_name:svchost.exe").aggregation("device_id")
+
+    pretty_response = {"results": []}
+    for event in query:
+        pretty_response["results"].append(event._info)
+
+    pprint(pretty_response)
+
+    print("\nCompare results manually with Postman")
+    print("----------------------------------------------------------")
+
+
+def main():
+    """Script entry point"""
+    global ORG_KEY
+    global HOSTNAME
+    parser = build_cli_parser()
+    parser.add_argument("--window", help="Define search window", default='3d')
+
+    args = parser.parse_args()
+    print_detail = args.verbose
+    window = '-' + args.window
+
+    if print_detail:
+        print("args provided {}".format(args))
+
+    cb = get_cb_cloud_object(args)
+    HEADERS["X-Auth-Token"] = cb.credentials.token
+    ORG_KEY = cb.credentials.org_key
+    HOSTNAME = cb.credentials.url
+
+    print()
+    print(17 * " ", "Enriched Events For Alerts")
+    print(SYMBOLS * DELIMITER)
+    # get alert created in the last four weeks, because enriched events are kept for some time
+    query = cb.select(CBAnalyticsAlert).set_create_time(range="-4w")
+    if len(query) == 0:
+        return
+
+    item = query[0]
+    alert_id = item._info["legacy_alert_id"]
+    eevents = [event._info for event in item.get_events()]
+
+    job_id = start_search_job(alert_id)
+    wait_till_job_ready(job_id)
+    records = get_results_search_job(job_id)
+    assert records == eevents, "Actual: {}, Expected: {}".format(records, eevents)
+    print("Get events for CBAnalyticsAlert...........................OK")
+    async_eevents = [event._info for event in item.get_events(async_mode=True).result()]
+    assert records == async_eevents, "Actual: {}, Expected: {}".format(records, async_eevents)
+    print("Get async events for CBAnalyticsAlert.....................OK")
+
+    # TODO rewrite the functionality below
```

Comment:
The rest of the main is just copy paste from src/tests/uat/process_search_calls.py, disregard for now.